File: mobile/android/app/src/main/java/tr/gov/ade/features/dashboard/DashboardViewModel.kt
package tr.gov.ade.features.dashboard

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import tr.gov.ade.core.network.APIClient
import tr.gov.ade.data.model.*
import javax.inject.Inject

/**
 * Dashboard ViewModel for ADE Android
 *
 * Handles dashboard data fetching and state management
 *
 * @author ADE Mobile Team
 * @since 2026-01-24
 */

@HiltViewModel
class DashboardViewModel @Inject constructor(
    private val apiClient: APIClient
) : ViewModel() {

    // MARK: - State

    private val _stats = MutableStateFlow<DashboardStats?>(null)
    val stats: StateFlow<DashboardStats?> = _stats.asStateFlow()

    private val _revenueData = MutableStateFlow<List<RevenueDataPoint>>(emptyList())
    val revenueData: StateFlow<List<RevenueDataPoint>> = _revenueData.asStateFlow()

    private val _recentOrders = MutableStateFlow<List<Order>>(emptyList())
    val recentOrders: StateFlow<List<Order>> = _recentOrders.asStateFlow()

    private val _aiSuggestions = MutableStateFlow<List<AISuggestion>>(emptyList())
    val aiSuggestions: StateFlow<List<AISuggestion>> = _aiSuggestions.asStateFlow()

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()

    private val _error = MutableStateFlow<String?>(null)
    val error: StateFlow<String?> = _error.asStateFlow()

    private val _selectedPeriod = MutableStateFlow(0)
    val selectedPeriod: StateFlow<Int> = _selectedPeriod.asStateFlow()

    // MARK: - Data Loading

    fun loadDashboard() {
        viewModelScope.launch {
            _isLoading.value = true

            try {
                // Load all data in parallel
                launch { loadStats() }
                launch { loadRecentOrders() }
                launch { loadRevenueChart() }
                launch { loadAISuggestions() }

            } catch (e: Exception) {
                _error.value = e.message ?: "Bilinmeyen bir hata oluştu"
            } finally {
                _isLoading.value = false
            }
        }
    }

    private suspend fun loadStats() {
        apiClient.safeApiCall {
            apiClient.service.getDashboardStats()
        }.onSuccess { response ->
            _stats.value = response
        }.onFailure { error ->
            _error.value = error.message
        }
    }

    private suspend fun loadRecentOrders() {
        apiClient.safeApiCall {
            apiClient.service.getECommerceStats()
        }.onSuccess { response ->
            _recentOrders.value = response.recentOrders
        }.onFailure { error ->
            _error.value = error.message
        }
    }

    private suspend fun loadRevenueChart() {
        val period = when (_selectedPeriod.value) {
            0 -> "7d"
            1 -> "30d"
            2 -> "12m"
            else -> "7d"
        }

        apiClient.safeApiCall {
            apiClient.service.getRevenueAnalytics(mapOf("period" to period))
        }.onSuccess { response ->
            _revenueData.value = response.byDate.map { dateRevenue ->
                RevenueDataPoint(
                    date = dateRevenue.date,
                    revenue = dateRevenue.revenue.toDouble()
                )
            }
        }.onFailure { error ->
            _error.value = error.message
        }
    }

    private suspend fun loadAISuggestions() {
        apiClient.safeApiCall {
            apiClient.service.getECommerceStats()
        }.onSuccess { response ->
            _aiSuggestions.value = response.aiSuggestions
        }.onFailure { error ->
            _error.value = error.message
        }
    }

    // MARK: - Actions

    fun changePeriod(period: Int) {
        _selectedPeriod.value = period
        viewModelScope.launch {
            loadRevenueChart()
        }
    }

    fun refresh() {
        loadDashboard()
    }

    fun clearError() {
        _error.value = null
    }
}

File: mobile/ios/ADE/Features/Dashboard/DashboardViewModel.swift
import Foundation
import Combine

/**
 * Dashboard ViewModel for ADE iOS
 *
 * Handles dashboard data fetching and state management
 *
 * @author ADE Mobile Team
 * @since 2026-01-24
 */

@MainActor
class DashboardViewModel: ObservableObject {
    // MARK: - Published Properties

    @Published var stats: DashboardStats?
    @Published var revenueData: [RevenueDataPoint] = []
    @Published var recentOrders: [Order] = []
    @Published var aiSuggestions: [AISuggestion] = []

    @Published var isLoading = false
    @Published var showError = false
    @Published var errorMessage: String?

    @Published var selectedPeriod = 0 {
        didSet {
            Task {
                await loadRevenueChart()
            }
        }
    }

    // MARK: - Dependencies

    private let apiClient = APIClient.shared
    private var cancellables = Set<AnyCancellable>()

    // MARK: - Initialization

    init() {
        // Subscribe to network state changes
        apiClient.$error
            .sink { [weak self] error in
                if let error = error {
                    self?.errorMessage = error.errorDescription
                    self?.showError = true
                }
            }
            .store(in: &cancellables)
    }

    // MARK: - Data Loading

    func loadDashboard() async {
        isLoading = true

        do {
            async let statsTask = loadStats()
            async let ordersTask = loadRecentOrders()
            async let chartTask = loadRevenueChart()
            async let suggestionsTask = loadAISuggestions()

            _ = try await (statsTask, ordersTask, chartTask, suggestionsTask)

        } catch {
            errorMessage = error.localizedDescription
            showError = true
        }

        isLoading = false
    }

    private func loadStats() async throws {
        let response: DashboardStats = try await apiClient.request(
            endpoint: .getDashboardStats
        )

        stats = response
    }

    private func loadRecentOrders() async throws {
        let response: ECommerceStats = try await apiClient.request(
            endpoint: .getECommerceStats
        )

        recentOrders = response.recentOrders
    }

    private func loadRevenueChart() async throws {
        // Determine period
        let period: String
        switch selectedPeriod {
        case 0: period = "7d"
        case 1: period = "30d"
        case 2: period = "12m"
        default: period = "7d"
        }

        let response: RevenueAnalytics = try await apiClient.request(
            endpoint: .getRevenueAnalytics,
            headers: ["period": period]
        )

        // Convert to chart data
        revenueData = response.byDate.map { dateRevenue in
            RevenueDataPoint(
                date: ISO8601DateFormatter().date(from: dateRevenue.date) ?? Date(),
                revenue: Double(truncating: dateRevenue.revenue as NSDecimalNumber)
            )
        }
    }

    private func loadAISuggestions() async throws {
        let response: ECommerceStats = try await apiClient.request(
            endpoint: .getECommerceStats
        )

        aiSuggestions = response.aiSuggestions
    }

    // MARK: - Actions

    func refresh() async {
        await loadDashboard()
    }
}

// MARK: - Helper Models

struct RevenueDataPoint: Identifiable {
    let id = UUID()
    let date: Date
    let revenue: Double
}
